lix/source-code/client/src/openRepository.ts

```diff
@@ -73,16 +78,38 @@ export async function openRepository(
 	const dir = "/"
 
 	let pending: Promise<void | { error: Error }> | undefined = clone({
-		fs: withLazyFetching(rawFs, "clone"),
-		http,
+		fs: withLazyFetching({ nodeishFs: rawFs, verbose, description: "clone" }),
+		http: makeHttpClient({
+			verbose,
+			description: "clone",
+
+			onReq: ({ url, body }: { url: string; body: any }) => {
+				return optimizedRefsReq({ url, body, addRef: args.branch })
```

Comment:
This is the call where multiple body manipulations will be required later on (i was referring to this in client.ts)